Update campaign list to support Evergreen campaigns and display CTR

JorgeMucientes opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #12389
This requires these FluxC changes to be merged first: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3086

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR adds the following design to Blaze campaign items: 

<img width="300"  src="https://github.com/user-attachments/assets/9d340b00-300c-49bd-a95a-29cd2fa963f0"> 

The logic of how campaign stats are displayed follows these rules: 

- Active (aka `active`, `scheduled` and `InModeration`) campaigns with end date will show “Remaining” budget
- Active Evergreen/Endless campaigns will show  weekly budget
- Finished campaigns will display “Total” spent budget

**EXTRA**: The [prior PR](https://github.com/woocommerce/woocommerce-android/pull/12324) to support Evergreen campaign missed a change to be able to successfully create this type of campaigns. I added that change here as it seemed overkill to create a PR just for it: https://github.com/woocommerce/woocommerce-android/pull/12433/commits/c488eafb631f12c375eae5333d550355de664452

### Testing information
1. Log into a site with Blaze enabled
2. Create an Evergreen campaign and check the budget is displayed as `weekly`
3. Create a campaign with end date and check the budget displays `remaining` while the campaign is active
4. Cancel both campaigns
5. Check Evergreen campaign still displays the `weekly` label
6. Check end date campaigns display `total` label now. 

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
I've performed the tests described in the instructions above. 
Smoked tested Blaze campaign creation flow.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->

https://github.com/user-attachments/assets/7fb87b6a-208a-4104-9aba-e511eccadec3

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->12389

JorgeMucientes committed:
Update FluxC changeset

JorgeMucientes committed:
Adds new isEndlessCampaign to Blaze model to support Evergreen campaigns

JorgeMucientes committed:
Merge branch 'refs/heads/trunk' into issue/12389-campaing-list-evergreen-support

# Conflicts:
#	build.gradle

JorgeMucientes committed:
Adds new isEndlessCampaign to Blaze model to support Evergreen campaigns

JorgeMucientes committed:
Update blaze campaign item to display new CTR and budget stats

JorgeMucientes committed:
Update FluxC changeset

JorgeMucientes committed:
Use the correct arrow symbol and fix UI glitches

JorgeMucientes committed:
Set the correct budget mode when creating evergreen campaigns

JorgeMucientes committed:
When budget mode is daily divide the total amount by weekly value

JorgeMucientes committed:
Merge branch 'refs/heads/trunk' into issue/12389-campaing-list-evergreen-support

# Conflicts:
#	build.gradle

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

JorgeMucientes committed:
Update FluxC changeset

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/notification/BlazeCampaignsObserver.kt:
```diff
@@ -39,7 +39,7 @@ class BlazeCampaignsObserver @Inject constructor(
     }
 
-    private fun scheduleNotification(campaigns: List<BlazeCampaignsDao.BlazeCampaignEntity>) {
+    private fun scheduleNotification(campaigns: List<BlazeCampaignModel>) {
```
Unrelated to the PR changes. It is a small side effect of updating `BlazeCampaignsStore.observeBlazeCampaigns` to return the domain model instead of the DB entity which shouldn't be propagated outside the storage layer.

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/budget/BlazeCampaignBudgetViewModel.kt:
```diff
@@ -39,7 +40,6 @@ class BlazeCampaignBudgetViewModel @Inject constructor(
 ) : ScopedViewModel(savedStateHandle) {
     companion object {
         const val MAX_DATE_LIMIT_IN_DAYS = 60
-        const val WEEKLY_DURATION = 7 // Used to calculate weekly budget in endless campaigns
```
Moved this to the `BlazeRepository` where all the constants belong

JorgeMucientes committed:
Fix unit tests compile issues

JorgeMucientes committed:
Test endless campaign request is built with correct budget values

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/BlazeUiMapper.kt:
```diff
@@ -0,0 +1,44 @@
+        stats = listOf(
+            BlazeCampaignStat(
+                name = R.string.blaze_campaign_status_ctr,
+                value = "$impressions ➔ $clicks"
```
I think we should move ➔ to the `strings.xml` file for localization. Otherwise, → will show the wrong direction in RTL languages.

JorgeMucientes replied:
Good catch!! Extracted the arrow to strings.xml 👍🏼

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/campaigs/BlazeCampaignItem.kt:
```diff
@@ -65,12 +66,19 @@ fun BlazeCampaignItem(
                             statName = stringResource(id = it.name),
                             statValue = it.value
                         )
+                        if (it != campaign.stats.last()) {
+                            Spacer(modifier = Modifier.weight(1f))
+                        }
```
I duplicated `campaign.stats` and made it four items. The UI doesn't look good. I think we should display only the first two items of `campaign.stats` in the UI. Otherwise, there’s a possibility that the backend might break the app's UI.

<img src="https://github.com/user-attachments/assets/8f2b0c41-18c4-4242-aa5e-6cb697cc34f6" width=300>

JorgeMucientes replied:
I'm not really concerned about this one (even though it's "fixed now). The number of stats we display is up to us. It doesn't depend on any number of stats returned by the backend. The Campaign stat concept is something specific to the Woo app code. The reason we were using `campaign.stats` array is because with the prior UI we depending on where `BlazeCampaignItem` was displayed (Dashboard card or campaign list) the number of stats would be 2 or 3. But now we display the same 2 stats everywhere, so I simplified de code. 

Hope this helps clarify a bit the prior implementation 🙂

irfano reviewed (changes requested):
Good job! I added two minor comments.

irfano reviewed (approved):
@JorgeMucientes, my requested changes are not blockers. I'm approving the PR now in case you want to address them in a separate PR and merge this feature for 20.2 before the codefreeze.

irfano commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/campaigs/BlazeCampaignItem.kt:
```diff
@@ -65,12 +66,19 @@ fun BlazeCampaignItem(
                             statValue = it.value
                         )
+                        if (it != campaign.stats.last()) {
+                            Spacer(modifier = Modifier.weight(1f))
```
I think setting spaces' weight to 1 does not work good here. You can see it in the horizontal mode. What about using fixed width for the space?

<img src="https://github.com/user-attachments/assets/54f21c8f-ad82-4499-88e2-681e0bed577a" width=600>

JorgeMucientes commented:
Thanks for the review and feedback @irfano, I will proceed to address it now 👍🏼

JorgeMucientes committed:
Extract arrow symbol to localized strings

JorgeMucientes committed:
Update FluxC changeset

JorgeMucientes commented:
I fixed the main issues about the arrow symbol not being localised 👍🏼 . Moving that string to `string.xml` res required a bit of refactoring: https://github.com/woocommerce/woocommerce-android/pull/12433/commits/ee27a142b1c15ed3a74b31ad0b420752e7897997. While I was at it I fixed the other 2 UI glitches. However, one of them wasn't real issue, I'll provide context about it in the specific comment.
